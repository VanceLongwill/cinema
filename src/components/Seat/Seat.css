.Seat {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: var(--secondary-light-color);
  padding: var(--spacing-sm);
  transition: 0.25s ease-out;
  overflow: hidden;
  border: 0.5rem solid transparent;
  color: var(--primary-light-color);
}

.Seat:not(.Seat-disabled):hover {
  cursor: pointer;
  transform: scale(1.1);
  background-color: var(--primary-highlight-color);
}

.Seat-disabled {
  background-color: var(--primary-dark-color);
}

.Seat-selected {
  border-color: var(--positive-color);
}

.SeatIcon {
  position: absolute;
  color: var(--primary-light-color);
}
