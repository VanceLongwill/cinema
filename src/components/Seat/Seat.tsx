import React from "react";
import cx from "classnames";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCouch } from "@fortawesome/free-solid-svg-icons";
import "./Seat.css";

export interface SeatProps {
  active: boolean;
  selected: boolean;
  onClick?: (id: string) => void;
  id: string;
}

export const Seat: React.FunctionComponent<SeatProps> = ({
  children,
  active,
  id,
  selected,
  onClick
}) => {
  const handleClick = React.useCallback(() => {
    if (active && onClick) {
      onClick(id);
    }
  }, [id, onClick, active]);
  return (
    <div
      className={cx({
        Seat: true,
        "Seat-disabled": !active,
        "Seat-selected": active && selected
      })}
      onClick={handleClick}
    >
      <FontAwesomeIcon className="SeatIcon" icon={faCouch} size="lg" />
      {children}
    </div>
  );
};
