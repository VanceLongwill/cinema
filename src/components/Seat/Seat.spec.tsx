import * as React from "react"

import { mount } from "enzyme"

import { Seat } from "./Seat"

describe("Seat", () => {
  it("should render without errors", () => {
    expect(mount(<Seat />)).toMatchSnapshot()
  })
})
