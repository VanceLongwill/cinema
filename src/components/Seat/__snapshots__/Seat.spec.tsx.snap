// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seat should render without errors 1`] = `
<Seat>
  <div
    className="Seat Seat-disabled"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="SeatIcon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            640,
            512,
            Array [],
            "f4b8",
            "M160 224v64h320v-64c0-35.3 28.7-64 64-64h32c0-53-43-96-96-96H160c-53 0-96 43-96 96h32c35.3 0 64 28.7 64 64zm416-32h-32c-17.7 0-32 14.3-32 32v96H128v-96c0-17.7-14.3-32-32-32H64c-35.3 0-64 28.7-64 64 0 23.6 13 44 32 55.1V432c0 8.8 7.2 16 16 16h64c8.8 0 16-7.2 16-16v-16h384v16c0 8.8 7.2 16 16 16h64c8.8 0 16-7.2 16-16V311.1c19-11.1 32-31.5 32-55.1 0-35.3-28.7-64-64-64z",
          ],
          "iconName": "couch",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="lg"
      spin={false}
      symbol={false}
      title=""
      transform={null}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-couch fa-w-20 fa-lg SeatIcon"
        data-icon="couch"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M160 224v64h320v-64c0-35.3 28.7-64 64-64h32c0-53-43-96-96-96H160c-53 0-96 43-96 96h32c35.3 0 64 28.7 64 64zm416-32h-32c-17.7 0-32 14.3-32 32v96H128v-96c0-17.7-14.3-32-32-32H64c-35.3 0-64 28.7-64 64 0 23.6 13 44 32 55.1V432c0 8.8 7.2 16 16 16h64c8.8 0 16-7.2 16-16v-16h384v16c0 8.8 7.2 16 16 16h64c8.8 0 16-7.2 16-16V311.1c19-11.1 32-31.5 32-55.1 0-35.3-28.7-64-64-64z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </FontAwesomeIcon>
  </div>
</Seat>
`;
